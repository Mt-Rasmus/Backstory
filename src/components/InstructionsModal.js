
import React from 'react';
import Modal from 'react-modal';
if (process.env.NODE_ENV !== 'test') Modal.setAppElement('#app');

const InstructionsModal = (props) => {
   return (
      <Modal
         isOpen={props.showModal}
         onRequestClose={props.handleToggleModal} // clicking esc or outside modal
         closeTimeoutMS={200}
         className="modal"
      >
         <p>
            These are the instructions
         </p>
         <button className="button" onClick={props.handleToggleModal}>
            Close
         </button>
      </Modal>
   )
}

export default InstructionsModal;