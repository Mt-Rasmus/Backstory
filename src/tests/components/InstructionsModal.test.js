
import React from 'react';
import InstructionsModal from '../../components/InstructionsModal';
import { shallow } from 'enzyme';

test('Should render Header correctly', () => {
   const wrapper = shallow(<InstructionsModal />);
   expect(wrapper).toMatchSnapshot();
});
