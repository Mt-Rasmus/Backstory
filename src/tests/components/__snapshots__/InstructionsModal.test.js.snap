// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render Header correctly 1`] = `
<Modal
  ariaHideApp={true}
  bodyOpenClassName="ReactModal__Body--open"
  className="modal inner-modal modal-wide"
  closeTimeoutMS={200}
  isOpen={false}
  parentSelector={[Function]}
  portalClassName="ReactModalPortal"
  role="dialog"
  shouldCloseOnEsc={true}
  shouldCloseOnOverlayClick={true}
  shouldFocusAfterRender={true}
  shouldReturnFocusAfterClose={true}
  style={
    Object {
      "overlay": Object {
        "backgroundColor": "rgba(0,0,0,0.25)",
      },
    }
  }
>
  <h1
    className="header__text"
  >
    How to Play
  </h1>
  <p
    className="modal__text-big"
  >
    This game can be played with two or more people.                 One person will act as game leader. This person selects a story and reads its premise                 aloud to everyone (click one of the icons on the Stories page).                 The game leader then reads the backstory to him-/herself without revealing it to the group                 (click the button labeled Backstory).                 The rest of the group then have to ask questions to the game leader                 in order to figure out what the backstory is.                 They may only ask 'yes' or 'no' questions.                 The game leader may reply to questions where incorrect assumptions have been made                 with: 'You've made an incorrect assumption'.                 If someone in the group is too scared to play, then please go easy on them.                 Just take a deep breath and try to understand that everyone may not have been blessed with                 your level of coolness.
  </p>
  <button
    className="button button--standard"
  >
    Close
  </button>
</Modal>
`;
